web/src/components/watch.tsx

```diff
@@ -37,12 +42,64 @@ export default function Watch(props: { name: string }) {
 
 	const play = () => usePlayer()?.play()
 
+	const toggleCatalog = () => {
+	    setShowCatalog(prev => {
+		const newState = !prev;
+		const catalogBox = document.getElementById("catalogBox");
+
+		if (catalogBox) {
+			catalogBox.style.display = newState ? 'block' : 'none';
+			const downloadButton = catalogBox.querySelector("#downloadButton") as HTMLButtonElement;
+			if(downloadButton) {
+				downloadButton.style.display = newState ? 'block' : 'none';
+			} else {
+				console.warn("Download button not found.");
+			}
+		} else {
+			console.warn("Catalog box not found.");
+		}
+
+		return newState;
+	    });
+	};
+
+	createEffect(() => {
+	    const player = usePlayer();
+	    const catalogBox = document.getElementById("catalogBox");
+	    if (player && catalogContainer && catalogBox) {
+			const downloadButton = catalogBox.querySelector("#downloadButton") as HTMLButtonElement;
+			if (showCatalog()) {
+				player.displayCatalog();
+				catalogBox.style.display = 'block';
+				if (downloadButton) {
+					downloadButton.style.display = 'block';
+				}
+			} else {
+				catalogBox.style.display = 'none';
+				if (downloadButton) {
+					downloadButton.style.display = 'none';
+				}
+			}
+	    }
+	});
+
 	// NOTE: The canvas automatically has width/height set to the decoded video size.
 	// TODO shrink it if needed via CSS
 	return (
 		<>
 			<Fail error={error()} />
 			<canvas ref={canvas} onClick={play} class="aspect-video w-full rounded-lg" />
+		    	<div class="flex mt-2">
+                		<button class="p-2 bg-blue-500 text-white rounded mr-2" style="font-size: 0.75rem; padding: 0.25rem 0.5rem;" onClick={toggleCatalog}>
+					{showCatalog() ? 'Hide Catalog' : 'Show Catalog'}
+                		</button>
+            		</div>
+        		<div id="catalogBox" style="position: relative; border: 1px solid #ccc; padding: 10px; margin-top: 10px; display: none; max-height: 300px; max-width: 100%;">
+            			<button id="downloadButton" class="p-2 bg-blue-500 text-white rounded" style="position: absolute; top: 10px; right: 10px; z-index: 10;font-size: 0.75rem; padding: 0.25rem 0.5rem;">
```

Comment:
Oof this CSS is pretty poor. Do you want me to fix it?

Reply:
Thank you for the feedback, appreciate your offer to help, I have attempted to make some refinements which include shifting the inline style to a separate `watch.css` file and modifying a few parameters to improve the appearance. Please advise if there are any further improvements to be made.